### DISCUSSION TITLE
Unit conversion

### DISCUSSION THREAD

**Discussion by TheSkorm** [miaowware/qrm2#403] [Ideas]
As a user I would like to convert between metric and imperial units.

A bot should have command like `?convert 23 meters to feet`, ideal it would be nice if it was smart enough to work with other syntax like `?convert 23m to f` or `?convert 23c to f`

An alternative solution would be to run for president in the US to change laws around common units, wait several decades before imperial falls out of use from common parlance, however as I am not born in the US it would have to be done by someone else.

This would aid during conversations between people of different countries

> why do yanks still use imperial units when talking about radio stuff when radio stuff is all in metric?
> its confusing af

**Comment by classabbyamp** [comment]
> An alternative solution would be to run for president in the US to change laws around common units, wait several decades before imperial falls out of use from common parlance, however as I am not born in the US it would have to be done by someone else.

Pres Carter tried that in the 70s... it passed, then nothing happened.

This is a good idea. Could implement the *nix program `units`.

**Comment by 0x5c** [comment]
After seeing how arby's (other bot) implemented `units`, and just how bad that command is that input interpretation, I'd kinda prefer something that has a python library, even if it meant making a wrapper for a C library.

**Comment by 0x5c** [comment]
Potential solution: https://pint.readthedocs.io/en/stable/index.html

**Comment by classabbyamp** [comment]
that library might also be better than the current `?dbconv` implementation
